### PR TITLE
Allowed ignoring scalar coordinates in `multi_model_statistics`

### LINKED ISSUE #1606: Preprocessor `multimodel` fails due to different scalar aux_coords
The creation of a preprocessor to compute global spatial averaged fields on native grids
correctly create 1D cubes but with different lat/lon (or i,j) values.
This trigger an iris exception error when the `multimodel` preprocessor attempt to merge the cubes
https://github.com/ESMValGroup/ESMValCore/blob/be6f654f059d5e07e0ab45ed4b90d92be6ad495d/esmvalcore/preprocessor/_multimodel.py#L293

As temporary workaround, using  data regridding before the spatial average solves the problem, 
but I think that the support to work with native grids data derivation is a relevant item for the system

**Attachments**
[test_time_mm.yml](https://github.com/ESMValGroup/ESMValCore/files/8796717/test_time_mm.yml.txt)
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/8796705/main_log_debug.txt)

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR allows the optional keyword `ignore_scalar_coords` for the `multi_model_statistics` and `ensemble_statistics` preprocessors. With this setting enabled, different scalar coords in the input cubes are ignored and do not raise merge `MergeError`s.

Closes #1606

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/_recipe/check.py [hunk 1]
```diff
@@ -251,6 +251,13 @@ def _verify_keep_input_datasets(keep_input_datasets):
             f"Must be defined as a boolean. Got {keep_input_datasets}.")
 
 
+def _verify_ignore_scalar_coords(ignore_scalar_coords):
+    if not isinstance(ignore_scalar_coords, bool):
+        raise RecipeError(
+            "Invalid value encountered for `ignore_scalar_coords`."
+            f"Must be defined as a boolean. Got {ignore_scalar_coords}.")
```
Comment (valeriupredoi) [inline esmvalcore/_recipe/check.py:258]:
```suggestion
            f"Must be defined as a boolean (true or false). Got {ignore_scalar_coords}.")
```

File: esmvalcore/preprocessor/_multimodel.py [hunk 2]
```diff
@@ -354,11 +354,16 @@ def _equalise_coordinate_metadata(cubes):
             # Note: remaining differences will raise an error at a later stage
             coord.long_name = None
 
-        # Additionally remove specific scalar coordinates which are not
-        # expected to be equal in the input cubes
-        scalar_coords_to_remove = ['p0', 'ptop']
+        # Remove scalar coordinates if desired. In addition, always remove
+        # specific scalar coordinates which are not expected to be equal in the
+        # input cubes.
+        scalar_coords_to_always_remove = ['p0', 'ptop']
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_multimodel.py:360]: can you add this to the documentation (up above in `preprocessor.rst`) pls

File: tests/integration/recipe/test_check.py [hunk 3]
```diff
@@ -310,6 +318,15 @@ def test_invalid_multi_model_keep_input():
         'Must be defined as a boolean. Got wrong.')
 
 
+def test_invalid_multi_model_ignore_scalar_coords():
+    with pytest.raises(RecipeError) as rec_err:
+        check._verify_ignore_scalar_coords(
+            INVALID_MM_SETTINGS['ignore_scalar_coords'])
+    assert str(rec_err.value) == (
+        'Invalid value encountered for `ignore_scalar_coords`.'
+        'Must be defined as a boolean. Got wrong.')
```
Comment (valeriupredoi) [inline tests/integration/recipe/test_check.py:327]:
```suggestion
        'Must be defined as a boolean (true or false). Got wrong.')
```

**Review by valeriupredoi — APPROVED** [review]
very cool, Manu! Just a couple of very minor review points from me :beer: Would we want to extend that [p0, ptop] list of coords to be always removed? If so, I'd put that at the top of the mm script, for now, maybe we'll take it out from there and expose it to the API some time in the future. About boolean (true or false) - it's always nice to tell the users stuff in layman terms, you know :grin:

**Comment by schlunma** [comment]
Thanks for the quick review @valeriupredoi!

> Would we want to extend that [p0, ptop] list of coords to be always removed? If so, I'd put that at the top of the mm script, for now, maybe we'll take it out from there and expose it to the API some time in the future.

I don't think we should expose this to the public API, especially with that new option now. We can also extend that list in there :+1:

**Comment by valeriupredoi** [comment]
> Thanks for the quick review @valeriupredoi!
> 
> > Would we want to extend that [p0, ptop] list of coords to be always removed? If so, I'd put that at the top of the mm script, for now, maybe we'll take it out from there and expose it to the API some time in the future.
> 
> I don't think we should expose this to the public API, especially with that new option now. We can also extend that list in there +1

Perfect then! Even less knobs and buttons for users to deal with it - I like it. Great work on this, hope signor Lovato is 'appy now, after we've left him out for dry for such a long time :grin:

**Comment by tomaslovato** [comment]
thanks a lot for the quick response @schlunma & @valeriupredoi

**Comment by valeriupredoi** [comment]
@ESMValGroup/technical-lead-development-team this is niice (in Borat accent) and ready for merging - please have a look and merge at earliest convenience, if you dillydally about it, myself or @remi-kazeroni will merge (given tight schedule on 2.8 freeze)

**Comment by zklaus** [comment]
It's a little bit muddled, isn't it? The documentation states
> To ignore *different* scalar coordinates in the input datasets, [...]

and
> *Differing* scalar coordinates can be ignored with the option ignore_scalar_coords.

but it's not quite clear what "ignore" really means and the code just removes all scalar coordinates.
This is stated correctly down in the notes on the API docs, where it says
> If True, remove any scalar coordinate in the input datasets before merging the input cubes into the multi-dataset cube. [...]

In lieu of implementing the behavior suggested by the documentation, i.e. removing only the differing scalar coordinates, I suggest clarifying the documentation in the two places mentioned above to make it easier to understand for the user what is happening.

**Comment by valeriupredoi** [comment]
bit of a :chicken: - :egg: problem, @zklaus -> ignoring stuff in input cubes means said stuff is not present in output cube(s) - that sounds very logical to me as @schlunma wrote it, but I would indeed change:

To ignore different scalar coordinates in the input datasets, use the option
``ignore_scalar_coords: true``.

with

To ignore all scalar coordinates in the input datasets, use the option
``ignore_scalar_coords: true``; this way these scalar coordinated will not be saved to the output multimodel cube.

**Comment by schlunma** [comment]
@zklaus I tried to improve the documentation, could you take another look? Thanks!
